examples/src/main/java/io/grpc/examples/manualflowcontrol/ManualFlowControlServer.java

```diff
@@ -0,0 +1,150 @@
+        final ServerCallStreamObserver<HelloReply> serverCallStreamObserver =
+            (ServerCallStreamObserver<HelloReply>) responseObserver;
+        serverCallStreamObserver.disableAutoInboundFlowControl();
+        // Signal the request sender to send one message.
+        serverCallStreamObserver.request(1);
+
+        // Set up a back-pressure-aware producer for the response stream. The onReadyHandler will be invoked
+        // when the consuming side has enough buffer space to receive more messages.
+        //
+        // Note: the onReadyHandler is invoked by gRPC's internal thread pool. You can't block in this in
+        // method or deadlocks can occur.
+        serverCallStreamObserver.setOnReadyHandler(new Runnable() {
+          @Override
+          public void run() {
+            // Start generating values from where we left off on a non-gRPC thread.
+            pool.execute(new Runnable() {
+              @Override
+              public void run() {
+                // requestStream.isReady() will go false when the consuming side runs out of buffer space and
+                // signals to slow down with back-pressure.
+                while (serverCallStreamObserver.isReady() && !serverCallStreamObserver.isCancelled()) {
+                  if (!work.isEmpty()) {
+                    // Send more messages if there are more messages to send.
+                    String name = work.remove();
+                    if (name != COMPLETED) {
+                      // Simulate doing some work to generate the next value
+                      try {
+                        Thread.sleep(200);
+                      } catch (InterruptedException e) {
+                        e.printStackTrace();
+                      }
+
+                      String message = "Hello " + name;
+                      System.out.println("<-- " + message);
+                      HelloReply reply = HelloReply.newBuilder().setMessage(message).build();
+
+                      // Send a response.
+                      responseObserver.onNext(reply);
+                    } else {
+                      System.out.println("Done.");
+                      responseObserver.onCompleted();
+                    }
+                  }
+                }
+              }
+            });
+          }
+        });
+
+        // Give gRPC a StreamObserver it can write incoming requests into.
+        return new StreamObserver<HelloRequest>() {
+          @Override
+          public void onNext(HelloRequest request) {
+            // Process the request and send a response or an error.
+            try {
+              // Accept and enqueue the request.
+              String name = request.getName();
+              System.out.println("--> " + name);
+              work.add(name);
+              // Signal the sender to send another request.
+              serverCallStreamObserver.request(1);
```

Comment:
I included `disableAutoInboundFlowControl()` on the server for illustration. It's paired with lines 54 and 109, where `request(1)` is called.

Technically, the scenario presented here could be implemented automatic flow control, but this demo's purpose is to show how to do everything manually.